mongodb/snooty
# Dop 3799: updating the URL with search input/filter

### Stories/Links:

DOP-3799

### Current Behavior:

https://www.mongodb.com/docs/search/?q=shell&searchProperty=realm-master

### Staging Links:

https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/landing/bianca.laube/DOP-3799/search/?q=hello&searchProperty=bi-connector-current&searchVersion=2.14

### Notes:
#### Changes
- now when input is entered on the persistent search bar it will update the URL accordingly 
- if a filter is selected, it will update the URL in both the prod and feature flag with the filter category and version
    - this was done since if you clicked on a search result and then want to go back to your filtered list it wouldn't store what your filter was in history but now it does 
#### Testing
The following list is something I have been using to keep track of testing the website. can be used as a guide for when reviewing:
1. adding a new input into the persistent search bar, make sure the URL changes
2. add a filter to the search, make sure the URL changes
3. choose a filter that has multiple versions, change the versions and make sure the URL updates
4. leave the filter but update the search input, make sure the URL updates
5. click on one of the search results, then go back and make sure the list is still correctly filtered
   - note: this doesnt really work on the stagging link since it keeps getting rid of the index.html in the URL so if you want to test this you have to do it locally

Extra test to run to make sure the other tickets in this project are still working properly:
1. Empty search on top nav search loads empty page
- https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/landing/bianca.laube/DOP-3791/search/index.html
- https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/landing/bianca.laube/DOP-3791/search/index.html?q=
2. Search of "failed%20to%20find%20a%20plan"* loads to a no results found modal
- [https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/landing/bianca.laube/DOP-3791/search/index.html?q="failed%20to%20find%20a%20plan](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/landing/bianca.laube/DOP-3791/search/index.html?q=%22failed%20to%20find%20a%20plan)"*
3. Search from the top nav bar shell (or right into the url) to see if it loads properly
- https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/landing/bianca.laube/DOP-3791/search/index.html?q=shell
4. Make a new search from the persistent search bar of with the input $and make sure it loads properly
5. Add a filter make sure skeleton shows
5. change the filter version and make sure the the skeleton loads again
6. Add a filter that shows no results (for example: the input $and with the filter atlas data lake
7. Do an empty search from the persistent search bar and make sure it shows the no results modal
8. run locally and switch the .env.development file for the feature flag off version and make sure all still looks good
